lib/methsel.g

```diff
@@ -70,3 +70,165 @@ NEXT_VMETHOD_PRINT_INFO := function ( methods, i, arity)
     Print("#I Trying next: ", methods[offset+4],
           " at ", methods[offset+5][1], ":", methods[offset+5][2], "\n");
 end;
+
+#############################################################################
+##
+#F  NewKeyBasedOperation( <name>, <requirements> )
+#F  DeclareKeyBasedOperation( <name>, <requirements> )
+#F  InstallKeyBasedMethod( <oper>[, <key>], <meth> )
```

Comment:
I am slightly concerned that people will confuse this with `KeyDependentOperation` resp. confuse one with the other. So if someone has a good idea for an alternate name...